rest_framework/compat.py

```diff
@@ -41,6 +41,13 @@ def distinct(queryset, base):
     uritemplate = None
 
 
+# async_to_sync is required for async view support
+if django.VERSION >= (4, 1):
+    from asgiref.sync import async_to_sync
+else:
+    async_to_sync = None
```

Comment:
How about

```
def async_to_sync(*args, **kwargs):
    raise NotImplementedError("DRF async only supports Django >= 4.1")
```

Reply:
A type checker would not be happy about this.

Reply:
`async_to_sync` is available in Django since version 3.1. In this case I'm using it just to run the test. If pytest-asyncio is added then it can be removed.

Reply:
I would personally rather we take a testing dependency on `pytest-asyncio` rather than implement a test-only compat helper.

Reply:
I believe it's best to defer the decision to add an additional dependency to project leads. Especially if the new dependency can be easily circumvented, such as in this case.

Reply:
As an a user of drf I would not happy to have any async tests related dependency in my project